### PR TITLE
CI/CD Pipeline False Positives: `import-dir` Exits with Code 0 on Failures

### LINKED ISSUE #473: CI/CD Pipeline False Positives: `import-dir` Exits with Code 0 on Failures
### Description
The `import-dir` command scans a directory and attempts to import all matching files. If one or more files fail to import, it records the failure in `ImportResult.FailedCount` and prints an error message. However, the command runner does not check the failed count to return a non-zero exit code.

### Impact
In CI/CD environments (GitHub Actions, Tekton, GitLab CI), the CLI exits with status `0`, indicating success even if all API specifications failed to import. This hides deployment failures.

### Code Reference
* [cmd/importDir.go](file:///c:/Users/Hp/microcks-cli/cmd/importDir.go#L162-L200)

### Diagram
```mermaid
graph TD
    A[Start import-dir] --> B[Scan Directory]
    B --> C[Import File 1: Success]
    B --> D[Import File 2: FAILED]
    C & D --> E[Display Results Summary]
    E --> F["Exit Status Code 0 (Success)"]
    F --> G["CI/CD pipeline reports green/success despite failures"]
```

### Steps to Reproduce
1. Run `microcks import-dir ./specs` where at least one spec has invalid content.
2. The console displays `✗ Failed: ...`.
3. Check the exit code of the process (`echo $?` or `$LASTEXITCODE` in PowerShell). It returns `0`.

---

### PR DESCRIPTION
**Pull request by aniket866**
### Issue
```mermaid
graph TD
    A[Start import-dir] --> B[Scan Directory]
    B --> C[Import File 1: Success]
    B --> D[Import File 2: FAILED]
    C & D --> E[Display Results Summary]
    E --> F["Exit Status Code 0 (Success)"]
    F --> G["CI/CD pipeline reports green/success despite failures"]
```

### Steps to Reproduce
1. Run `microcks import-dir ./specs` where at least one spec has invalid content.
2. The console displays `✗ Failed: ...`.
3. Check the exit code of the process (`echo $?` or `$LASTEXITCODE` in PowerShell). It returns `0`.

---

fixed: #473 

@Harishrs2006 @lbroudoux 
Please do review

### CODE REVIEW

**Comment by aniket866** [comment]
@Vaishnav88sk do checkout this one too

File: cmd/importDir.go [hunk 1]
```diff
@@ -197,6 +197,9 @@ func NewImportDirCommand(globalClientOpts *connectors.ClientOptions) *cobra.Comm
 			}
 
 			fmt.Printf("\nImport completed: %d/%d files imported successfully\n", result.SuccessCount, result.TotalFiles)
+			if result.FailedCount > 0 {
+				os.Exit(1)
+			}
```
Comment (Vaishnav88sk) [inline cmd/importDir.go:196]: This is the right behavior change for CI/CD, but please add a test that covers the partial-failure path and verifies we return a non-zero exit when `FailedCount > 0`.

**Review by Vaishnav88sk — CHANGES_REQUESTED** [review]

File: cmd/importDir.go (hunk 1 above)
Comment (aniket866) [reply cmd/importDir.go:196]: hi @Vaishnav88sk I have added the respective tests

**Comment by Vaishnav88sk** [comment]
LGTM
@Harsh4902

**Comment by Caesarsage** [comment]
Thank you @aniket866 

This has been Fixed on master, import-dir now returns a non-zero exit when any file fails: importDirectoryPartialFailure wraps FailedCount > 0 into a KindAPI error that the exit funnel maps to a failure code. 

I believe this can be closed.
